### ISSUE TITLE
Omit breaks in 3.9.2

### ISSUE THREAD

**Issue by zombieJ** [microsoft/TypeScript#38531]
**TypeScript Version:** 
3.9.2

**Search Terms:** 
3.9.2 Omit

**Code**

```ts
interface A { 
  my: 'aaa';
}

interface B { 
  my: 'bbb';
}

interface C extends Omit<A & B, 'my'> { 
  my: string;
}
```

**Expected behavior:**

Can define `my` with Omited like prev version.

**Actual behavior:**

> Property 'my' of type 'string' is not assignable to string index type 'never'.(2411)

**Playground Link:** 

https://www.typescriptlang.org/play/?ssl=1&ssc=1&pln=12&pc=1#code/JYOwLgpgTgZghgYwgAgILIN7ILAChnIC2AngFzIDkc1FA3HgL556iSyIoBCmO+RZlAEbC6jZrlbR4SZAGFkEAB6QQAEwDOyAPKFgYADzoAZMk4AaSiQoA+HngIly6sFFABzeria4gA

**Related Issues:**

**Comment by kitsonk** [comment]
Effectively duplicate of #38241, related to breaking change #37195.

**Comment by ahejlsberg** [comment]
The issue here is that `A & B` is an empty intersection (i.e. a type for which you couldn't possibly construct a value) that now reduces to `never`. We previously didn't catch such types.

You could instead write your code as

```ts
type Base = Omit<A, 'my'> & Omit<B, 'my'>;

interface C extends Base { 
  my: string;
}
```

That is, omit the conflicting discriminants _before_ you intersect the types.

**Comment by zombieJ** [comment]
@ahejlsberg , but never can be omit:
```tsx
interface A { 
  my: never;
}

interface C extends Omit<A, 'my'> { 
  my: string;
}
```

https://www.typescriptlang.org/play/?ssl=1&ssc=1&pln=8&pc=1#code/JYOwLgpgTgZghgYwgAgILIN7ILAChnIC2AngFzIgQBu0A3HgL556iSyIoDCyEAHpCAAmAZ2QB5QsDAAeVABpkAchKKAfJhz4iZZMLBRQAc3q4muIA

**Comment by ahejlsberg** [comment]
Yes, but any type that includes a property of type `never` is a type with _no_ possible values, so we're not talking about something that couldn't exist in the real world. Ideally we'd reduce all such types to `never` (that is, instead of `{ my: never }` we'd simply have `never`) but it is costly to check for and tricky to do in a lazily evaluated type system.
